### PR TITLE
Use lib-iterate and lib-log throughout the provisioner

### PR DESCRIPTION
**Pull request by jhford**
Basically, switch over iteration logic to lib-iterate and the logging stuff from debug to lib-log.  I spent a lot of time looking into making the logs nicer.  They're sooooo much nicer to read.  I also removed a bunch of those places where I was intercepting exceptions to log them before rethrowing them.

npm-shrinkwrap throws off the github numbers, but in case you're interested:

 18 files changed, 362 insertions(+), 839 deletions(-)

Here's an example of a complete provisioning iteration where we submit two spot requests:
```
16:54:40.196Z  INFO aws-provisioner: provisioning iteration starting
16:54:40.534Z  INFO aws-provisioner: loaded worker types
16:54:40.534Z  INFO aws-provisioner: starting state update for all regions
16:54:40.977Z  INFO aws-provisioner: ran all state promises for region (region=eu-central-1)
16:54:40.978Z  INFO aws-provisioner: found maximum prices (region=eu-central-1)
16:54:42.465Z  INFO aws-provisioner: ran all state promises for region (region=us-west-2)
16:54:42.466Z  INFO aws-provisioner: found maximum prices (region=us-west-2)
16:54:46.369Z  INFO aws-provisioner: ran all state promises for region (region=us-west-1)
16:54:46.370Z  INFO aws-provisioner: found maximum prices (region=us-west-1)
16:54:46.677Z  INFO aws-provisioner: ran all state promises for region (region=us-east-1)
16:54:46.678Z  INFO aws-provisioner: found maximum prices (region=us-east-1)
16:54:46.678Z  INFO aws-provisioner: finished state update for all regions
16:54:46.678Z  INFO aws-provisioner: available availability zones
    zones: [
      "eu-central-1a",
      "eu-central-1b",
      "us-east-1a",
      "us-east-1c",
      "us-east-1d",
      "us-east-1e",
      "us-west-1b",
      "us-west-1c",
      "us-west-2a",
      "us-west-2b",
      "us-west-2c"
    ]
16:54:46.678Z  INFO aws-provisioner: finished all aws state update operations
16:54:46.678Z  INFO aws-provisioner: updated aws state
16:54:49.026Z  INFO aws-provisioner: obtained bias info
16:54:49.026Z  INFO aws-provisioner: resources tagged
16:54:49.026Z  INFO aws-provisioner: rogue resources killed
16:54:49.026Z  INFO aws-provisioner: zombie resources killed
16:54:49.988Z  INFO aws-provisioner: configured worker types
    workerTypes: [
      "cli"
    ]
16:54:50.681Z  INFO aws-provisioner: got pending tasks count (workerType=cli, pendingTasks=0)
16:54:50.681Z  INFO aws-provisioner: desired change would be lower than minimum capacity (runningCapacity=0, pendingCapacity=0, pendingTasks=0, idealChange=0, actualChange=2)
16:54:50.682Z  INFO aws-provisioner: determined change (workerType=cli, change=2)
16:54:51.349Z  INFO aws-provisioner: wrote state to azure (workerType=cli)
16:54:51.351Z  INFO aws-provisioner: starting to submit spot requests
16:54:51.351Z  INFO aws-provisioner: starting to submit spot requests in region (region=us-west-2)
16:54:51.351Z  INFO aws-provisioner: asking to spawn instance
16:54:52.281Z  INFO aws-provisioner: created secret (token=<snip>)
16:54:53.337Z  INFO aws-provisioner: submitted spot request (srid=sir-03cjegw5, price=0.1262, workerType=cli, region=us-west-2, zone=us-west-2a, instanceType=r3.xlarge)
16:54:53.338Z  INFO aws-provisioner: asking to spawn instance
16:54:53.509Z  INFO aws-provisioner: created secret (token=<snip>)
16:54:54.757Z  INFO aws-provisioner: submitted spot request (srid=sir-03clbxf2, price=0.1262, workerType=cli, region=us-west-2, zone=us-west-2a, instanceType=r3.xlarge)
16:54:54.758Z  INFO aws-provisioner: finished submiting spot requests in region (region=us-west-2)
16:54:54.758Z  INFO aws-provisioner: finished submiting spot requests
16:54:54.758Z  INFO aws-provisioner: provisioning iteration complete (duration=14562)
```

### CODE REVIEW

**Comment by jhford** [comment]
@gregarndt not expecting a review from you, just wanted to give you an example of what the logging changes look like

File: src/main.js [hunk 1]
```diff
@@ -266,19 +278,73 @@ let load = base.loader({
...
+          apiKey: cfg.deadmanssnitch.api.key,
+          snitchUrl: cfg.deadmanssnitch.iterationSnitch,
+        },
+        handler: async (watchdog, state) => {
+          // Store the stats somewhere
+          if (!state.stats) {
+            state.stats = {
+              runs: 0,
+              consecFail: 0,
+              overallFail: 0,
```
Comment (jonasfj) [inline src/main.js]: why?

**Comment by gregarndt** [comment]
Does it always log in that format or does it log in that format when a terminal is attached, and then maybe spit out json to parse when it's logged to some centralized logging system?

**Comment by gregarndt** [comment]
I definitely like this style of logging, which is similar to what we do in docker-worker where there is an "event" string describing the message, and then some key value stuff afterwards

**Comment by jonasfj** [comment]
this is a lot of changes... I gave it a quick look through and disregarded most of the logging changes... the approach seems sane.

r+

**Comment by jhford** [comment]
@gregarndt so, right now, it's just in that format.  WHen I finish the other side of structured logging, it'll dump out in pure json, then be interpreted by something into human readable.

**Comment by jhford** [comment]
This got merged in.

**Pull request closed without merging**